src/test/scala/chiselTests/experimental/hierarchy/Utils.scala

```diff
@@ -0,0 +1,69 @@
+
+import chisel3._
+import _root_.firrtl.annotations._
+import chisel3.stage.{ChiselCircuitAnnotation, CircuitSerializationAnnotation, DesignAnnotation}
+import chiselTests.ChiselRunners
+import firrtl.stage.FirrtlCircuitAnnotation
+import org.scalatest.matchers.should.Matchers
+
+trait Utils extends ChiselRunners with chiselTests.Utils with Matchers {
+  import Annotations._
+  def check(bc: => RawModule, target: IsMember, tag: String, printOutput: Boolean = false): Unit = {
+    val (output, annotations) = getFirrtlAndAnnos(bc)
+    if (printOutput) println(output.serialize)
+    val anno = MarkAnnotation(target, tag)
+    assert(annotations.contains(anno), s"${annotations} does not contain $anno!")
+  }
+  def check(bc: => RawModule, instance: String, of: String): Unit = {
+    val (output, _) = getFirrtlAndAnnos(bc)
+    val insts = output.serialize.split("\n").map(_.split("\\s+").toList).collect {
+      case _ :: "inst" :: x :: "of" :: y :: _ => (x, y)
+    }
+    assert(insts.contains((instance, of)), s"""Output does not contain $instance of $of! Contains the following:${"\n"}${insts.mkString("\n")}""")
+  }
+  def check(bc: => RawModule, targets: Seq[(IsMember, String)], lines: Seq[String]): Unit = {
+    val (res, annos) = getFirrtlAndAnnos(bc)
+    val chirrtl = res.serialize
+    val annotations = annos
+      .filter {
+        case a: ChiselCircuitAnnotation => false
+        case a: CircuitSerializationAnnotation => false
+        case a: FirrtlCircuitAnnotation => false
+        case a: DesignAnnotation[_] => false
+        case a: DeletedAnnotation => false
+        case a: _root_.firrtl.options.Unserializable => false
+        case _ => true
+      }
+    for ((target, tag) <- targets) {
+      val anno = MarkAnnotation(target, tag)
+      annotations should contain(anno)
+    }
+    for (line <- lines) {
+      chirrtl should include(line)
+    }
+  }
+  def check(bc: => RawModule, aspect: Annotation, checker: PartialFunction[Annotation, Boolean]): Unit = {
+    val (res, annos) = getFirrtlAndAnnos(bc, Seq(aspect))
+    val annotations = annos
+      .flatMap {
+        case a: ChiselCircuitAnnotation => Nil
+        case a: CircuitSerializationAnnotation => Nil
+        case a: FirrtlCircuitAnnotation => Nil
+        case a: DesignAnnotation[_] => Nil
+        case a: DeletedAnnotation => Nil
+        case a: _root_.firrtl.options.Unserializable => Nil
+        case a => Seq(a)
+      }
+    annotations.collectFirst(checker) should contain(true)
+  }
+  // TODO promote to standard API (in FIRRTL) and perhaps even implement with a macro
+  implicit class Str2RefTarget(str: String) {
+    def rt: ReferenceTarget = Target.deserialize(str).asInstanceOf[ReferenceTarget]
+    def it: InstanceTarget = Target.deserialize(str).asInstanceOf[InstanceTarget]
+    def mt: ModuleTarget = Target.deserialize(str).asInstanceOf[ModuleTarget]
+    def ct: CircuitTarget = Target.deserialize(str).asInstanceOf[CircuitTarget]
+  }
+}
```

Comment:
This is nitty, but using overloaded methods to hide the Scalatest checking (in a separate file) makes the tests really hard to parse. I'd much prefer if every test included the "test" that it was doing.

E.g., I'd expect to see all tests have some some "assert" or "foo should include/contain bar" statement.

I understand that this is done to try and keep the tests tidy, but I think there's a middle ground here.